tests/agent_equipment/test_validator.py

```
@@ -129,6 +133,215 @@ def test_file_loading_rejects_ambiguous_and_non_json_documents(self) -> None:
+
+            with patch.object(validator.os, "open", side_effect=record_open):
+                result = load_catalog_lock(catalog_path, lock_path)
+
+        self.assertIsNotNone(result.model)
+        self.assertEqual(len(opened_descriptors), 2)
+        self.assertTrue(
+            all(flags & os.O_NOFOLLOW for flags in open_flags),  # type: ignore[attr-defined]
+        )
+        self.assertTrue(all(flags & os.O_NONBLOCK for flags in open_flags))
+        for descriptor in opened_descriptors:
+            with self.assertRaises(OSError):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)